pre_commit/languages/julia.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import annotations
+
+import contextlib
+from collections.abc import Generator
+from collections.abc import Sequence
+
+from pre_commit import lang_base
+from pre_commit.envcontext import envcontext
+from pre_commit.envcontext import PatchesT
+from pre_commit.prefix import Prefix
+from pre_commit.util import cmd_output_b
+
+ENVIRONMENT_DIR = 'juliaenv'
+health_check = lang_base.basic_health_check
+get_default_version = lang_base.basic_get_default_version
+
+
+def run_hook(
+        prefix: Prefix,
+        entry: str,
+        args: Sequence[str],
+        file_args: Sequence[str],
+        *,
+        is_local: bool,
+        require_serial: bool,
+        color: bool,
+) -> tuple[int, bytes]:
+    # `entry` is a (hook-repo relative) file followed by (optional) args, e.g.
+    # `bin/id.jl` or `bin/hook.jl --arg1 --arg2` so we
+    # 1) shell parse it and join with args with hook_cmd
+    # 2) prepend the hooks prefix path to the first argument (the file)
+    # 3) prepend `julia` as the interpreter
+    cmd = lang_base.hook_cmd(entry, args)
+    cmd = ('julia', prefix.path(cmd[0]), *cmd[1:])
```

Comment:
I suspect this will need special handling for `is_local: True` -- otherwise hooks won't be able to use `julia` with local scripts -- `language: r` is very similar to this so perhaps some inspiration can be drawn from that? (although `r` has a lot more complicated setup so I guess just look at run part for inspiration)

Reply:
Yes, thanks for the pointer. Added a test.